src/git-town-append

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+source "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/helpers/helpers.sh"
+
+
+function ensure_has_target_branch {
+  if [ -z "$target_branch_name" ]; then
+    echo_error_header
+    echo_error "No branch name provided."
+    exit_with_error newline
+  fi
+}
+
+
+function preconditions {
+  target_branch_name=$1
+  ensure_has_target_branch
+
+  parent_branch_name=$INITIAL_BRANCH_NAME
+  if [ "$parent_branch_name" != "$MAIN_BRANCH_NAME" ]; then
+    ensure_knows_parent_branches "$parent_branch_name"
+  fi
+
+  if [ "$(has_remote_url)" = true ]; then
+    fetch
+  fi
+
+  ensure_does_not_have_branch "$target_branch_name"
+
+  export RUN_IN_GIT_ROOT=true
+  export STASH_OPEN_CHANGES=true
+}
+
+
+function steps {
+
+  # sync the current branch
```

Comment:
Good point, I have removed the `parent_branch_variable`. Let's do the refactor of outputting the steps to sync all ancestor branches into a dedicated function separately. I have toyed with the idea a number of times, but given that this is a very meta-function, and it would add a lot of abstraction for negligible savings of code lines, have shied away from it.